src/AspNetCore/Src/filewatcher.cxx

```diff
@@ -126,7 +155,8 @@ Win32 error
         cbCompletion = 0;
     }
 
-    return 0;
+    pFileMonitor->m_fThreadExit = TRUE;
+    ExitThread(0);
```

Comment:
Why are we calling ExitThread instead of returning here?

Reply:
just want to make sure the thread signal/status does change.  https://msdn.microsoft.com/en-us/library/windows/desktop/ms682659(v=vs.85).aspx, as we call waitforsingleobject.